editor/plugins/skeleton_3d_editor_plugin.cpp

```diff
@@ -1363,6 +1335,39 @@ void Skeleton3DGizmoPlugin::redraw(EditorNode3DGizmo *p_gizmo) {
 		selected = se->get_selected_bone();
 	}
 
+	Ref<ArrayMesh> m = get_bones_mesh(skeleton, selected, p_gizmo->is_selected());
+	p_gizmo->add_mesh(m, Ref<Material>(), Transform3D(), skeleton->register_skin(skeleton->create_skin_from_rest_transforms()));
+}
+
+Ref<ArrayMesh> Skeleton3DGizmoPlugin::get_bones_mesh(Skeleton3D *skeleton, int selected, bool is_selected) {
+	Ref<StandardMaterial3D> unselected_mat = Ref<StandardMaterial3D>(memnew(StandardMaterial3D));
+	unselected_mat->set_shading_mode(StandardMaterial3D::SHADING_MODE_UNSHADED);
+	unselected_mat->set_transparency(StandardMaterial3D::TRANSPARENCY_ALPHA);
+	unselected_mat->set_flag(StandardMaterial3D::FLAG_ALBEDO_FROM_VERTEX_COLOR, true);
+	unselected_mat->set_flag(StandardMaterial3D::FLAG_SRGB_VERTEX_COLOR, true);
+
+	Ref<ShaderMaterial> selected_mat = Ref<ShaderMaterial>(memnew(ShaderMaterial));
+	Ref<Shader> selected_sh = Ref<Shader>(memnew(Shader));
```

Comment:
The shaders must be stored somewhere since it is bad way to memnew them every time.